storage/netapp/netapp_e_storagepool.py

```diff
@@ -0,0 +1,888 @@
+        except:
+            err = get_exception()
+            pool_id = self.pool_detail['id']
+            self.module.exit_json(
+                msg="Failed to update reserve drive count. Pool id [%s]. Array id [%s].  Error[%s]." % (pool_id,
+                                                                                                        self.ssid,
+                                                                                                        str(
+                                                                                                            err)))
+
+    def apply(self):
+        changed = False
+        pool_exists = False
+
+        self.pool_detail = self.get_storage_pool(self.name)
+
+        if self.pool_detail:
+            pool_exists = True
+            pool_id = self.pool_detail['id']
+
+            if self.state == 'absent':
+                self.debug("CHANGED: storage pool exists, but requested state is 'absent'")
+                changed = True
+            elif self.state == 'present':
+                # sanity checks first- we can't change these, so we'll bomb if they're specified
+                if self.criteria_drive_type and self.criteria_drive_type != self.pool_detail['driveMediaType']:
+                    self.module.fail_json(
+                        msg="drive media type %s cannot be changed to %s" % (self.pool_detail['driveMediaType'],
+                                                                             self.criteria_drive_type))
+
+                # now the things we can change...
+                if self.needs_expansion:
+                    self.debug("CHANGED: storage pool needs expansion")
+                    changed = True
+
+                if self.needs_raid_level_migration:
+                    self.debug(
+                        "CHANGED: raid level migration required; storage pool uses '%s', requested is '%s'" % (
+                            self.pool_detail['raidLevel'], self.raid_level))
+                    changed = True
+
+                    # if self.reserved_drive_count_differs:
+                    # changed = True
+
+                    # TODO: validate other state details? (pool priority, alert threshold)
+
+                    # per FPoole and others, pool reduce operations will not be supported. Automatic "smart" reduction
+                    # presents a difficult parameter issue, as the disk count can increase due to expansion, so we
+                    # can't just use disk count > criteria_drive_count.
+
+        else:  # pool does not exist
+            if self.state == 'present':
+                self.debug("CHANGED: storage pool does not exist, but requested state is 'present'")
+                changed = True
+
+                # ensure we can get back a workable set of disks
+                # (doing this early so candidate selection runs under check mode)
+                self.disk_ids = self.get_candidate_disks()
```

Comment:
My ide made a similar recommendation but since I'm fairly new to python and ansible I decided to leave it.  I think any parameter that requires users to work with ids of any sort is a poor design.  A Tray/Drawer/Slot location identification scheme is better than straight ids from the proxy or box itself.  But the primary use case for this module is to leave the details of finding the right number and type of disk to the code (based on capacity and pool or group type).  So to directly answer your question, I don't think create_storage_pool can be called before the object is instantiated.  

I'd like to change this module to provide T/D/S but that is a fairly significant design change.  I didn't write these modules, I'm the guy who's making sure they work, removing errors and removing obvious technical debt.